### ISSUE TITLE
AutoRegisteringInputObjectGraphType excludes properties with only the get method

### ISSUE THREAD

**Issue by shumutun** [graphql-dotnet/graphql-dotnet#3551]
## Description

AutoRegisteringInputObjectGraphType excludes properties that are with only the get method but both serializers (SystemTextJson and NewtonsoftJson) can deserialize objects using the default constructor. 
if I design a data contract class that initializes via a constructor and then try to use this data contract to build an Input Type - the input type is created empty. 

The problem is in the GetRegisteredMembers() of the AutoRegisteringInputObjectGraphType<TSourceType> class. The 
`.Where(x => x.CanWrite)` is unnecessary.

### Steps to reproduce

1. Create a class that properties are initialized via a constructor only
2. Wrap this class into the AutoRegisteringInputObjectGraphType<TSourceType>

### Expected result

All properties of the class became Input Type fields

### Actual result

The result Input Type is empty

### Environment

doesn't matter

###  Workaround
Override the incorrect behavior:
```
public class AutoRegisteringInputObjectGraphTypeAllProperties<TSourceType> : AutoRegisteringInputObjectGraphType<TSourceType>
{
    private readonly Expression<Func<TSourceType, object?>>[]? _excludedProperties;

    protected AutoRegisteringInputObjectGraphTypeAllProperties()
        : this(null)
    {
    }

    protected AutoRegisteringInputObjectGraphTypeAllProperties(
        params Expression<Func<TSourceType, object?>>[]? excludedProperties)
        : base(excludedProperties)
    {
        _excludedProperties = excludedProperties;
    }

    protected override IEnumerable<MemberInfo> GetRegisteredMembers() => ExcludeProperties(
        typeof(TSourceType).GetProperties(BindingFlags.Public | BindingFlags.Instance), _excludedProperties);

    private static IEnumerable<PropertyInfo> ExcludeProperties(IEnumerable<PropertyInfo> properties,
        params Expression<Func<TSourceType, object?>>[]? excludedProperties)
    {
        if (excludedProperties == null || excludedProperties.Length == 0)
            return properties;
        return properties.Where(propertyInfo => excludedProperties.All(p => GetPropertyName(p) != propertyInfo.Name));
    }

    private static string GetPropertyName(Expression<Func<TSourceType, object?>> expression)
    {
        if (expression.Body is MemberExpression m1)
            return m1.Member.Name;

        if (expression.Body is UnaryExpression u && u.Operand is MemberExpression m2)
            return m2.Member.Name;

        throw new NotSupportedException($"Unsupported type of expression: {expression.GetType().Name}");
    }
}
```

**Comment by Shane32** [comment]
It was designed this way because, for an input type, the values are read from the GraphQL query and written to the input object.

Within `ParseDictionary` there is code that calls `ToObject` to convert the dictionary of values into `TSourceType`.  `ToObject` does support using constructors to initialize objects currently.  Note that the serializer is not involved in this process.

We can consider reviewing the behavior of `AutoRegisteringInputObjectGraphType` for v8.  I would have it examine the constructors to determine if a match can be found for the property name before allowing it to be automatically added.

Another option may be to add init-only properties.  Have you tried that?

**Comment by sungam3r** [comment]
Agree that we _can_ remove `CanWrite` requirement. Then let's add runtime check instead. So input type (TSourceType) should either have setter for that property or ctor with argument matching that name.

**Comment by Shane32** [comment]
With #3771 and #3773 being merged, this feature request is now a viable feature to add, and useful when dealing with record types.

See:
- #3771 
- #3773 

cc: @gao-artur

**Comment by Shane32** [comment]
This feature is included in v8 via #3782
